lib/resty/rediscluster.lua

```diff
@@ -106,14 +129,57 @@ local function check_auth(self, redis_client)
     end
 end
 
-local function release_connection(red, config)
-    local ok,err = red:set_keepalive(config.keepalive_timeout
-            or DEFAULT_KEEPALIVE_TIMEOUT, config.keepalive_cons or DEFAULT_KEEPALIVE_CONS)
-    if not ok then
-        ngx.log(ngx.ERR,"set keepalive failed:", err)
+
+local function check_status(self, redis_client)
+    if not redis_client then
+        return nil, "redis_client can not be nil"
+    end
+
+    local info, err = redis_client:cluster("info")
+    if err then
+        return nil, "failed to fetch cluster info: " .. err
+    end
+
+    local state = match(info, "cluster_state:(%a+)\r\n")
+    if not state then
+        return nil, "failed to match cluster state"
+    end
+
+    if state == "fail" then
+        return nil, "cluster state: fail"
+    end
+    if state == "ok" then
+        return true
+    end
+end
+
+
+local function check_version(self, redis_client)
```

Comment:
Looks like this function only checks the version of redis supported by this lib? And redis 8 has not been released yet, it seems that this function can drop

Reply:
Just ensure it a customer is using the latest version 8, and we are not yet ready. Giving them a message.

Reply:
I think this is an advanced optimization, it is on the hot path, I think indicating the supported version on README.md is enough.

Reply:
Currently, this function is only called when redis cluster is down or there is network issues. We can assume most of the time, it is not called.

By the way, let me update the readme.md as well.

Reply:
README.md updated.